compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -948,6 +958,7 @@ object Contexts {
     /** Sources and Files that were loaded */
     val sources: util.HashMap[AbstractFile, SourceFile] = util.HashMap[AbstractFile, SourceFile]()
     val files: util.HashMap[TermName, AbstractFile] = util.HashMap()
+    val siblingClassfiles: util.HashMap[AbstractFile, AbstractFile] = util.HashMap()
```

Comment:
I this cache based on referential equality?

Reply:
no, its based on `equals`, the reference equality one is `util.EqHashMap`

Reply:
What I meant to ask is if the `equals` of the sub-classes of `AbstractFile` are implemented or defaulting to referential equality.

Reply:
for PlainFile (nio Path based) its based on interning the absolute path string, for virtual files its reference equality, and Zip entries are virtual files, but are cached in their parent

Reply:
removed the cache in contexts